source/gateway.go

```diff
@@ -494,10 +495,10 @@ func gwRouteHasParentRef(routeParentRefs []v1.ParentReference, ref v1.ParentRefe
 	return false
 }
 
-func gwRouteIsAccepted(conds []metav1.Condition) bool {
+func gwRouteIsAccepted(conds []metav1.Condition, meta *metav1.ObjectMeta) bool {
 	for _, c := range conds {
 		if v1.RouteConditionType(c.Type) == v1.RouteConditionAccepted {
-			return c.Status == metav1.ConditionTrue
+			return c.Status == metav1.ConditionTrue && c.ObservedGeneration == meta.Generation
 		}
 	}
 	return false
```

Comment:
false case not covered with test

![Screenshot 2025-05-04 at 14 39 34](https://github.com/user-attachments/assets/ff36aa79-1cfb-49cc-8c65-cc9827c8a276)

Reply:
Just added a test for this as well, though that was existing code.

Reply:
Indeed. This code Existed. As we touched function, it's nice to help a project a bit ;-)